packages/table/src/tableHotkeys.ts

```diff
@@ -295,19 +303,39 @@ export class TableHotkeys {
             return;
         }
 
+        const frozenRowsHeight = this.grid.getCumulativeHeightBefore(this.state.numFrozenRowsClamped);
+        const frozenColumnsWidth = this.grid.getCumulativeWidthBefore(this.state.numFrozenColumnsClamped);
+
+        // We align the scrollable container with non-frozen cells and the focused cell at the
+        // same origin: the end of the frozen columns and rows (`O`).
+        //            → scrollLeft
+        //       → cumulative cell widths
+        // * +---VIEWPORT RECT---+
+        // * |   | H | H | H | H |
+        // * +---+---+---+---+---+ ↓ cumulative row heights
+        // * | H | F | F | F | F |
+        // * +---+---O--SCROL. C.+
+        // * | H | F | S | S | S | ↓ scrollTop
+        // * +---+---+---+---+---+
+        // * | H | F | S | S | S |
+        // * +---+---+---+---+---+
+        // * | H | F | S | S | S |
+        // * +---+---+---+---+---+
+
         // sort keys in normal CSS position order (per the trusty TRBL/"trouble" acronym)
         // tslint:disable:object-literal-sort-keys
-        const viewportBounds = {
+        const scrollableContainerBounds = {
             top: viewportRect.top,
-            right: viewportRect.left + viewportRect.width,
-            bottom: viewportRect.top + viewportRect.height,
+            right: viewportRect.left + viewportRect.width - frozenColumnsWidth - this.state.rowHeaderWidth,
+            bottom: viewportRect.top + viewportRect.height - frozenRowsHeight - this.state.columnHeaderHeight,
```

Comment:
it seems unexpected these would be subtracted from right and bottom since the frozen cells/headers are rendered in the top and left

Reply:
> it seems unexpected these would be subtracted from right and bottom since the frozen cells/headers are rendered in the top and left

I agree. This comes down to the fact that the offset of the scrollable part of `viewportRect` is with respect to the point `O` after headers and frozen cells (see: [ASCII table](https://github.com/palantir/blueprint/pull/6882/files#diff-8cd23c90b1959192d311f6112803042130c4c7709017e6cac6fd991ea89221beR309-R323)). So, `top` and `left` are, in fact, correct; we just need to adjust `bottom` and `right` to get the correct bounds of the scrollable container.

See also:

https://github.com/palantir/blueprint/pull/6882/files#r1659176907